src/chai/ManagedArray.inl

```diff
@@ -408,6 +422,13 @@ ManagedArray<T>::getActiveBasePointer() const
   return m_active_base_pointer;
 }
 
+template<typename T>
+PointerRecord const * 
+ManagedArray<T>::getPointerRecord() const
```

Comment:
What's the use case for this?

Reply:
In our wrapper class when we reallocate we create a new `ManagedArray` and move the data into it manually because our wrapper class can hold non POD data. We'd also like to transfer the user-callback to the new class. Getting the pointer record from the `ArrayManager` won't work if the `ManagedArray` hasn't been allocated yet, and in our use cases we set the call back before doing allocation.

Reply:
https://github.com/GEOSX/cxx-utilities/blob/fb749fdb6acff24edff49ac1b79d5c6c13a77d68/src/src/ChaiBuffer.hpp#L111